### PR TITLE
Traps disarming gives morale effects

### PR DESCRIPTION
**Pull request by AMurkin**
#### Summary
```SUMMARY: Features "Traps disarming gives morale effects"```

#### Purpose of change
Add the impact of success or failure of trap disarming.
Introduce generic morale types.

#### Describe the solution
Added generic morale types Accomplishment and Failure.
Apply `Accomplishment` morale buff on successful trap disarming. 
Apply `Failure` morale debuff on failed trap disarming.

#### Additional context
Don't know if morale effects are good inside the `map.cpp`.

### CODE REVIEW

File: data/json/morale_types.json [hunk 1]
```diff
@@ -329,6 +329,16 @@
     "type": "morale_type",
     "text": "Dug out a grave"
   },
+  {
+    "id": "morale_accomplishment",
+    "type": "morale_type",
+    "text": "Accomplishment"
```
Comment (Night-Pryanik) [inline data/json/morale_types.json]: This seems like to be a way too generic name. Imagine we add several reasons for this morale bonus to shoot, and the player could see several "Accomplishment"s in his morale display. I believe it needs to be specified to what type of accomplishment caused this morale boost.

**Comment by Rail-Runner** [comment]
I'm not sure it makes sense. Seems like it makes it possible for player to get morale boost from placing a trap and then disarming it immediately, which is weird.

**Comment by paulenka-aleh** [comment]
To me making a successful complex craft or construction (the one that could fail), successfully passing speech check with NPC and successful hacking seems to be more in line with what I feel than trap disarming

File: data/json/morale_types.json (hunk 1 above)
Comment (AMurkin) [reply data/json/morale_types.json]: It was intentional. We should use generic morale types and less of them instead of creating more of similar types.

**Comment by AMurkin** [comment]
> I'm not sure it makes sense. Seems like it makes it possible for player to get morale boost from placing a trap and then disarming it immediately, which is weird.

And you may fail and trigger it. Get a morale debuff and more nasty things.
Although I should greatly reduce values.

File: data/json/morale_types.json (hunk 1 above)
Comment (kevingranade) [reply data/json/morale_types.json]: There used to be a stacking issue, but now it's just a matter of avoiding clutter in the UI, which is still a pretty big deal.  The player is notified specifically what the accomplishment was when they perform the action, but minutes or hours later, does it matter what it was, or that they're happy about it?

**Comment by nexusmrsep** [comment]
I would also opt for reducing the bonus according to progression in traps skill by dividing the bonus by the players's skill. A professional trapper would barely notice the effects of success of disarming yet another trap, while disarming a landmine by a novice might be an achievement.

**Comment by AMurkin** [comment]
I agree. And what about negative morale effects? Should they increase with skill progression?

**Comment by AMurkin** [comment]
Rebased to resolve merge conflicts.

**Pull request merged**